Core/Core/Api/GraphQL/Resources/ActiveUserResource.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using GraphQL;
+using Speckle.Core.Api.GraphQL.Inputs;
+using Speckle.Core.Api.GraphQL.Models;
+using Speckle.Core.Api.GraphQL.Models.Responses;
+
+namespace Speckle.Core.Api.GraphQL.Resources;
+
+public sealed class ActiveUserResource
+{
+  private readonly ISpeckleGraphQLClient _client;
+
+  internal ActiveUserResource(ISpeckleGraphQLClient client)
+  {
+    _client = client;
+  }
+
+  /// <summary>
+  /// Gets the currently active user profile.
+  /// </summary>
+  /// <param name="cancellationToken"></param>
+  /// <returns></returns>
+  /// <returns>the requested user, or null if the user does not exist (i.e. <see cref="Client"/> was initialised with an unauthenticated account)</returns>
+  /// <inheritdoc cref="ISpeckleGraphQLClient.ExecuteGraphQLRequest{T}"/>
+  public async Task<UserInfo?> Get(CancellationToken cancellationToken = default)
+  {
+    //language=graphql
+    const string QUERY = """
+                          query User {
+                           activeUser {
+                             id,
+                             email,
+                             name,
+                             bio,
+                             company,
+                             avatar,
+                             verified,
+                             profiles,
+                             role,
+                           }
+                         }
+                         """;
+    var request = new GraphQLRequest { Query = QUERY };
```

Comment:
I don't think there's much value in unit tests for this. Our integration tests will already validate that the query is valid for the schema, and that's about all you could possibly unit test here...

But since this and all the other resource classes depend on the `ISpeckleGraphQLClient` interface, nothing is preventing us from mocking where the query goes... The GraphQLRequest is just a lightweight wrapper on the query + variables.

Reply:
If we have integration tests all good - do they run within the build?